src/Microsoft.AspNetCore.Authentication.JwtBearer/JwtBearerOptions.cs

```diff
@@ -118,5 +118,21 @@ public JwtBearerOptions() : base()
         /// <see cref="Http.Authentication.AuthenticationProperties"/> after a successful authorization.
         /// </summary>
         public bool SaveToken { get; set; } = true;
+
+        /// <summary>
+        /// Authentication failures with one of these exception types, or types derived from these, will have their messages
+        /// included in a 'X-Authentication-Failure' response header.
+        /// </summary>
+        public ICollection<Type> ReportableFailures { get; set; } = new List<Type>()
```

Comment:
Not a huge fan of a generic white list feature like this, I think for something like this, we should just hardcode this list and turn it into a custom error.  Or if we wanted to make it somewhat extensible, then I would rather something like an `Func<string, Exception>` transform, where the default instance would take these exception types and produce the appropriate error for them.